react-components/src/hooks/useClickedNode.test.tsx

```diff
@@ -0,0 +1,358 @@
+    const { result } = renderHook(() => useClickedNodeData(), { wrapper });
+
+    const clickCallback = getClickCallback(viewer);
+
+    await simulateClick(
+      clickCallback,
+      MOUSE.LEFT,
+      mockScreenPositionOnClick.x,
+      mockScreenPositionOnClick.y
+    );
+
+    await waitFor(async () => {
+      expect(result.current).not.toBeUndefined();
+      expect(result.current).toEqual({
+        mouseButton: MOUSE.LEFT,
+        position: mockScreenPositionOnClick,
+        fdmResult: undefined,
+        assetMappingResult: undefined,
+        pointCloudAnnotationMappingResult: mockPointCloudData,
+        pointCloudFdmVolumeMappingResult: null,
+        intersection: mockCadIntersection
+      });
+    });
+  });
+
+  test('returns point cloud FDM volume mapping result when intersection and data are available', async () => {
+    const mockVolumeData = [
+      { assetInstance: { space: 'test', externalId: 'volume-1' }, views: [] }
+    ];
+
+    viewerGetAnyIntersectionFromPixelMock.mockResolvedValue(mockCadIntersection);
+
+    mockUsePointCloudFdmVolumeMappingForIntersection.mockReturnValue(
+      createMockQueryResult(mockVolumeData, false)
+    );
+
+    const { result } = renderHook(() => useClickedNodeData(), { wrapper });
+
+    const clickCallback = getClickCallback(viewer);
+
+    await simulateClick(
+      clickCallback,
+      MOUSE.LEFT,
+      mockScreenPositionOnClick.x,
+      mockScreenPositionOnClick.y
+    );
+
+    await waitFor(async () => {
+      expect(result.current).not.toBeUndefined();
+      expect(result.current).toEqual({
+        mouseButton: MOUSE.LEFT,
+        position: mockScreenPositionOnClick,
+        fdmResult: undefined,
+        assetMappingResult: undefined,
+        pointCloudAnnotationMappingResult: null,
+        pointCloudFdmVolumeMappingResult: mockVolumeData,
+        intersection: mockCadIntersection
+      });
+    });
+  });
+
+  test('sets up event listeners for 360 annotation intersection handling', async () => {
+    const { result } = renderHook(() => useClickedNodeData(), { wrapper });
+
+    expect(viewer.on).toHaveBeenCalledWith('click', expect.any(Function));
+    expect(result.current).toBeUndefined();
+  });
+
+  test('returns undefined when no intersection data is available', () => {
+    const { result } = renderHook(() => useClickedNodeData(), { wrapper });
+
+    expect(result.current).toBeUndefined();
+  });
+
+  test('do nothing when left click is false', async () => {
+    viewerGetAnyIntersectionFromPixelMock.mockResolvedValue(mockCadIntersection);
+
+    const { result } = renderHook(() => useClickedNodeData({ leftClick: false }), { wrapper });
+
+    const clickCallback = getClickCallback(viewer);
+
+    await simulateClick(
+      clickCallback,
+      MOUSE.LEFT,
+      mockScreenPositionOnClick.x,
+      mockScreenPositionOnClick.y
+    );
+
+    expect(result.current).toBeUndefined();
+  });
+
+  test('do nothing when right click is true', async () => {
```

Comment:
The result in this test is not nothing, either the test is wrong or the description is wrong